code/drasil-code/Language/Drasil/Code/Imperative/GOOL/Symantics.hs

```diff
@@ -10,23 +10,26 @@ import Database.Drasil (ChunkDB)
 import Language.Drasil.Code.DataDesc (DataDesc)
 import Language.Drasil.CodeSpec (Comments, Verbosity)
 
-import GOOL.Drasil (ProgData)
+import GOOL.Drasil (ProgData, GOOLState)
 
+import Control.Monad.State (State)
 import Text.PrettyPrint.HughesPJ (Doc)
 
 class (AuxiliarySym repr) => PackageSym repr where
   type Package repr 
-  package :: ProgData -> [repr (Auxiliary repr)] -> repr (Package repr)
+  package :: State GOOLState ProgData -> [repr (Auxiliary repr)] -> 
```

Comment:
Likewise, passing in a `State` like this doesn't feel right.  If `package` was all in the `State` monad, that would be different.